File: AuctionMarket/src/main/java/com/portfolio/auctionmarket/domain/user/dto/UserProfileResponse.java
package com.portfolio.auctionmarket.domain.user.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.portfolio.auctionmarket.domain.sellers.entity.Seller;
import com.portfolio.auctionmarket.domain.user.entity.Role;
import com.portfolio.auctionmarket.domain.sellers.entity.SellerStatus;
import com.portfolio.auctionmarket.domain.user.entity.User;
import com.portfolio.auctionmarket.global.util.MaskingUtil;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class UserProfileResponse {
    private Long userId;
    private String email;
    private String username;
    private String phone;
    private String nickname;
    private String baseAddress;
    private String detailAddress;
    private Role role;
    private Long point;
    private String avgRating;
    private SellerStatus sellerStatus;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime updatedAt;

    public static UserProfileResponse from(User user){
        String formatPhone = MaskingUtil.formatPhone(user.getPhone());
        Seller seller = user.getSeller();
        SellerStatus sellerStatus = (seller != null) ? seller.getStatus() : SellerStatus.NONE;

        String detailAddress = user.getDetailAddress() != null ? user.getDetailAddress() : "";

        return UserProfileResponse.builder()
                .userId(user.getUserId())
                .email(user.getEmail())
                .username(user.getUsername())
                .phone(formatPhone)
                .nickname(user.getNickname())
                .baseAddress(user.getBaseAddress())
                .detailAddress(detailAddress)
                .role(user.getRole())
                .point(user.getPoint())
                .avgRating(String.format("%.1f", user.getAvgRating()))
                .sellerStatus(sellerStatus)
                .createdAt(user.getCreatedAt())
                .updatedAt(user.getUpdatedAt())
                .build();
    }
}

File: AuctionMarket/auction-app-user/src/main/kotlin/org/example/auctioncommon/domain/user/dto/UserProfileResponse.kt
package org.example.auctioncommon.domain.user.dto

import com.fasterxml.jackson.annotation.JsonFormat
import org.example.auctioncommon.domain.seller.entity.Seller
import org.example.auctioncommon.domain.seller.entity.SellerStatus
import org.example.auctioncommon.domain.user.entity.Role
import org.example.auctioncommon.domain.user.entity.User
import org.example.auctioncommon.global.util.MaskingUtil
import java.time.LocalDateTime

data class UserProfileResponse(
    val userId: Long,
    val email: String,
    val username: String,
    val phone: String,
    val nickname: String,
    val baseAddress: String,
    val detailAddress: String,
    val role: Role,
    val point: Long,
    val avgRating: String,
    val sellerStatus: SellerStatus,

    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private val createdAt: LocalDateTime? = null,

    @field:JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private val updatedAt: LocalDateTime? = null,
)

fun User.toProfileDto() : UserProfileResponse{
    val formatPhone: String? = MaskingUtil.formatPhone(phone)
    val seller: Seller? = seller
    val sellerStatus: SellerStatus? = if (seller != null) seller.status else SellerStatus.NONE
    val detailAddress: String? = if (detailAddress != null) detailAddress else ""
    val formatRating: String = String.format("%.1f", avgRating)

    return UserProfileResponse(
        userId = this.userId ?: 0L,
        email = this.email,
        username = this.username,
        phone = formatPhone.toString(),
        nickname = this.nickname,
        baseAddress = this.baseAddress,
        detailAddress = detailAddress.toString(),
        role = this.role,
        point = this.point,
        avgRating = formatRating,
        sellerStatus = sellerStatus ?: SellerStatus.NONE,
        createdAt = this.createdAt,
        updatedAt = this.updatedAt

    )
}